Functionality for fallback values and Nested selectors

rtrufin opened the pull request:
Test for Mixins that Nest Selectors #13.
Fixed issue with properties that have fallback.
E.g.: 

```
padding: 16px;
padding: 1rem;
```

Commit:
Support for mixin nested selectors & multiple values for same property

Test for Mixins that Nest Selectors #13.
Fixed issue with properties that have fallback.
E.g.: padding: 16px;
padding: 1rem;

Commit:
Update README.md

Commit:
updated readme with the new functionality

ryanbahniuk commented:
Thanks for the PR, will check this out this weekend.

ryanbahniuk commented on src/mixinResult.js:
```diff
@@ -123,6 +123,23 @@ MixinResult.prototype = {
+      declarationValues.push(declaration.value);
+    });
+    var message = 'Value: ' + declarationValues + ' does not equal value: ' + value + '.';
+    assert.equal(declarationValues.toString(), value.toString(), message);
```
I'd rather make the assertion with `deepEqual` than cast these to strings.

ryanbahniuk commented:
Can you add integration and unit tests for the new features?

rtrufin commented:
Sure, I'll start working on them this week, Also, I'll test the deepEqual assertion. Thanks for the reply

rtrufin commented:
I have some issues while running the tests, on the mixinResult and FuncResult test files:

```
FuncResult
    wrapFunctionWithArgs
      1) "before each" hook for "wraps the function and args with the necessary SCSS to not fail the compiler"

MixinResult
    wrapMixinWithArgs
      2) "before each" hook for "wraps the standalone mixin to not fail the compiler"
```

these two errors block all the tests from those files from running

ryanbahniuk commented:
I checked out your branch and got all the tests to pass on Node 0.12.0 and Node 5.11.0. However, it is possible that your problem is due to the Sass binaries failing to install during `npm install`. I would confirm you are on at least 0.12.0 and then delete your node modules folder, and retry installing the dependencies.

pvdlg commented:
Hello,
is there any plan to merge this PR? It would be really useful.
Thanks

rtrufin commented:
@vanduynslagerp (totally forgot about this PR) I'm going to work on it later this week. I've had the issues with the unit tests and I'm trying to see what I can do about that

razvantrufin committed:
Resolved tests that were failing

rtrufin commented:
Fixed the tests. The issue was caused by the paths that were not concatenated properly. Added the normalize-paths module to handle that issue.

rtrufin committed:
Merge branch 'master' into master

razvantrufin committed:
Locked node sass version to 3.1.2

razvantrufin committed:
Merge branches 'master' and 'master' of https://github.com/rtrufin/sassaby

razvantrufin committed:
Locked node sass version to 2.81.0

razvantrufin committed:
Locked node sass version to 2.0.1

razvantrufin committed:
Added request 2.81.0 module as a possible fix for node-sass issue

razvantrufin committed:
Added cross env to handle the set of the node_env environment variable

razvantrufin committed:
Added node_env value in travis config file

razvantrufin committed:
Added more node js versions to test

ryanbahniuk commented on src/sassaby.js:
```diff
@@ -5,66 +5,67 @@ var assert = require('assert');
+    for (var variableName in varz) {
+        sassVariables = sassVariables + '$' + variableName + ':' + varz[variableName] + ';';
+    }
+    return sassVariables;
```
Can you undo the indentation changes to this file so the diff reads cleaner?

ryanbahniuk commented on README.md:
````diff
@@ -333,6 +333,42 @@ Assert that the mixin makes a declaration of the given rule-property pair.
+}
+```
+```js
+sassaby.includedMixin('font-size').calledWithArgs('2').declaresProperties('max-height', ['32px', '2rem']);
````
I think I'd rather have this functionality in the current API. I could see `declares` taking either a string or an array. If it is an array with one item it will echo current behavior and if its an array with multiple items it will make the fallback assertion defined here.

rtrufin committed:
Fixed indentation

razvantrufin committed:
Fixed functionality & added unit tests to cover all the implementation

razvantrufin committed:
Removed newer versions of node

rtrufin commented:
I've fixed all the issues, and it seems that I also had some issues with the coverage. It is now at 100%. 
Also, I'll probably start working on another branch in order to make the module work on newer versions of node.

ryanbahniuk commented:
Looks good. Did you see my previous comment (got collapsed by GitHub) that I think the new `declaresProperties` behavior should actually just live in the existing `declares` method. I can see that method taking 3 types of arguments:

`string` - current behavior which will stay around for backwards compatibility
`array` - if it has 1 item it will have very similar behavior to the string. if it has more than 1 item it will assert on a fallback like you have in your `declaresProperties` method.

What do you think about that?

rtrufin commented:
@ryanbahniuk I've changed declares to work both with arrays and Strings and removed the declaresProperties method. See: [mixinResult.js](https://github.com/rtrufin/sassaby/blob/b14a0274710208b0a8b628f246a2c585bd724480/src/mixinResult.js)

ryanbahniuk commented:
@rtrufin looks good. Can you make sure the docs match the new updates?

rtrufin committed:
Update Readme to remove declaresProperties.

declaresProperties has been merged into the declares method
